test/dart2_9_suggestors/dart2_9_utilities_test.dart

```diff
@@ -0,0 +1,410 @@
+              (props) {},
+              _\$Foo, // ignore: undefined_identifier
+            );
+          ''');
+        });
+
+        test('when there are too many arguments', () {
+          _expectNullReturnValue('''
+            UiFactory<FooProps> Foo = uiForwardRef(
+              (props) {},
+              _\$FooConfig, // ignore: undefined_identifier
+              'another arg',
+            );
+          ''');
+        });
+      });
+
+      group('returns the generated factory config', () {
+        void _expectConfigName({String input, String expectedName}) {
+          final unit = parseString(content: input).unit;
+          final argList = (unit.declarations
+                  .whereType<TopLevelVariableDeclaration>()
+                  .first
+                  .variables
+                  .variables
+                  .first
+                  .initializer as MethodInvocation)
+              .argumentList;
+
+          final returnValue = getGeneratedFactoryConfigArg(argList);
+          expect(returnValue, isA<SimpleIdentifier>());
+          expect(returnValue.name, expectedName);
+        }
+
+        test('when the config is public', () {
+          _expectConfigName(
+            input: '''
+              UiFactory<FooProps> Foo = uiFunction(
+                (props) {},
+                \$FooConfig, // ignore: undefined_identifier
+              );
+            ''',
+            expectedName: '\$FooConfig',
+          );
+        });
+
+        test('when the config is private', () {
+          _expectConfigName(
+            input: '''
+              UiFactory<FooProps> Foo = uiFunction(
+                (props) {},
+                _\$FooConfig, // ignore: undefined_identifier
+              );
+            ''',
+            expectedName: '_\$FooConfig',
+          );
+        });
+
+        test('when the config has a number in the name', () {
+          _expectConfigName(
+            input: '''
+              UiFactory<FooProps> Foo = uiFunction(
+                (props) {},
+                \$Foo2Config, // ignore: undefined_identifier
+              );
+            ''',
+            expectedName: '\$Foo2Config',
+          );
+        });
+
+        test('when the config has an underscore in the name', () {
+          _expectConfigName(
+            input: '''
+              UiFactory<FooProps> _Foo = uiFunction(
+                (props) {},
+                \$_FooConfig, // ignore: undefined_identifier
+              );
+            ''',
+            expectedName: '\$_FooConfig',
+          );
+        });
+
+        test('when the config is used in uiForwardRef', () {
+          _expectConfigName(
+            input: '''
+              UiFactory<FooProps> Foo = uiForwardRef(
+                (props) {},
+                _\$FooConfig, // ignore: undefined_identifier
+              );
+            ''',
+            expectedName: '_\$FooConfig',
+          );
+        });
```

Comment:
Could we get a test here showing a `uiFunction` wrapped in `memo`?

Reply:
Added in 17b25b1